learning_observer/learning_observer/auth/social_sso.py

```diff
@@ -210,6 +248,34 @@ async def _process_student_documents(student):
             await _process_student_documents(student)
     # TODO saved skipped doc ids somewhere?
 
+async def _canvas(request):
+    '''
+    Handle Canvas authorization
+    '''
+    if 'error' in request.query:
+        return {}
+    
+    token_uri = settings.pmss_settings.token_uri(types=['lms', 'canvas_oauth'])
+    url = token_uri
+    
+    params = {
+        "grant_type": "refresh_token",
+        'client_id': settings.pmss_settings.client_id(types=['lms', 'canvas_oauth']),
+        'client_secret': settings.pmss_settings.client_secret(types=['lms', 'canvas_oauth']),
+        "refresh_token": settings.pmss_settings.refresh_token(types=['lms', 'canvas_oauth'])
+    }
+    async with aiohttp.ClientSession(loop=request.app.loop) as client:
+        async with client.post(url, data=params) as resp:
+            data = await resp.json()
+        assert 'access_token' in data, data
+
+        # get user profile
+        canvas_headers = {'Authorization': 'Bearer ' + data['access_token']}
+        session = await aiohttp_session.get_session(request)
+        session[constants.CANVAS_AUTH_HEADERS] = canvas_headers
+        request[constants.CANVAS_AUTH_HEADERS] = canvas_headers
+        
+    return data
 
 async def _google(request):
```

Comment:
I know this isn't your code, but while we're fixing horrible function names.....

Reply:
I've also fixed this function name too